UserService/src/main/java/com/beaconfire/userservice/controller/UserAuthController.java

```diff
@@ -88,4 +101,30 @@ public ResponseEntity<ChangePasswordResponse> changePassword(@RequestBody Change
                 .message("Password changed successfully.")
                 .build());
     }
+
+    /*****   email verification (newly added)  ********/
+    @PostMapping("/user/verify")
+    @PreAuthorize("hasAnyAuthority('ROLE_SADMIN')")
+    @Operation(summary = "Allows authenticated users to change their password",
+            responses = {
+                    @ApiResponse(responseCode = "200", description = "Password changed successfully",
+                            content = @Content(schema = @Schema(implementation = ChangePasswordResponse.class))),
+                    @ApiResponse(responseCode = "400", description = "Invalid email verify request",
+                            content = @Content(schema = @Schema(implementation = ErrorDetails.class))),
+            })
+    public ResponseEntity<EmailVerificationResponse> verifyEmail(@RequestBody EmailVerificationRequest emailVerificationRequest) {
+
+        if (emailService.verifyEmail(emailVerificationRequest.getToken())) {
+            return ResponseEntity.ok(EmailVerificationResponse.builder()
+                    .verified(true)
+                    .message("Email verified successfully!")
+                    .build());
+        } else {
+            return ResponseEntity.ok(EmailVerificationResponse.builder()
```

Comment:
Use ExceptionHandler to handle response with invalid request

Reply:
I think the request is valid although the token expired. So, for the frontend, only displays the result message.

Reply:
If a token is expired, then it is not a 200. Should throw exception at the service layer, then handle by the handler

Reply:
Okay I will change this latter